src/Files.App/Actions/Open/OpenInVSAction.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Files.App.Contexts;
+using Files.App.Shell;
+using Microsoft.Win32;
+
+namespace Files.App.Actions
+{
+	internal class OpenInVSAction : ObservableObject, IAction
+	{
+		private readonly IContentPageContext _context;
+
+		private readonly bool _isVSInstalled;
+
+		public string Label { get; } = "OpenInVS".GetLocalizedResource();
+
+		public string Description { get; } = "OpenInVSDescription".GetLocalizedResource();
+
+		public bool IsExecutable => 
+			_isVSInstalled &&
+			!string.IsNullOrWhiteSpace(_context.SolutionFilePath);
+
+		public OpenInVSAction()
+		{
+			_context = Ioc.Default.GetRequiredService<IContentPageContext>();
+
+			_isVSInstalled = IsVSInstalled();
+			if (_isVSInstalled )
+				_context.PropertyChanged += Context_PropertyChanged;
+		}
+
+		private void Context_PropertyChanged(object? sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == nameof(IContentPageContext.SolutionFilePath))
+				OnPropertyChanged(nameof(IsExecutable));
+		}
+
+		public Task ExecuteAsync()
+		{
+			Win32API.RunPowershellCommand($"start {_context.SolutionFilePath}", false);
+
+			return Task.CompletedTask;
+		}
+
+		private static bool IsVSInstalled()
```

Comment:
Small nit but it might be cleaner to create a service or helper for this kind of thing.

Reply:
Do you mean a service/helper to retrieve if any given program is installed? Or instead a service/helper that contains the two methods I created?

Reply:
I meant the two methods but it might be useful to expand it for all programs in the future.

Reply:
Done